spec/models/concerns/variant_stock_spec.rb

```diff
@@ -90,13 +90,33 @@
       end
     end
 
-    context 'when the variant has no stock item' do
+    context 'when the variant has not been saved yet' do
       let(:variant) { build(:variant) }
 
+      it 'has no stock items' do
+        expect(variant.stock_items.count).to eq 0
+      end
+
       it 'returns stock location default' do
         expect(variant.on_demand).to be_falsy
       end
     end
+
+    context 'when the variant has been soft-deleted' do
+      let(:deleted_variant) { create(:variant) }
```

Comment:
I think you could save the `before` block.

```suggestion
      let(:deleted_variant) { create(:variant).tap(&:destroy) }
```

Reply:
Done :+1: